src/net_processing.h

```diff
@@ -17,7 +17,10 @@ static const int64_t ORPHAN_TX_EXPIRE_TIME = 20 * 60;
 static const int64_t ORPHAN_TX_EXPIRE_INTERVAL = 5 * 60;
 /** Default number of orphan+recently-replaced txn to keep around for block reconstruction */
 static const unsigned int DEFAULT_BLOCK_RECONSTRUCTION_EXTRA_TXN = 100;
-
+/** Headers download timeout expressed in microseconds
+ *  Timeout = base + per_header * (expected number of headers) */
+static constexpr int64_t HEADERS_DOWNLOAD_TIMEOUT_BASE = 15 * 60 * 1000000; // 15 minutes
+static constexpr int64_t HEADERS_DOWNLOAD_TIMEOUT_PER_HEADER = 1000; // 1ms/header
```

Comment:
Given our headers are a LOT bigger than Bitcoin Core (because they contain a Litecoin header and a transaction), should this be higher? I think most of that time will be finding the header on disk, though, so thinking 1000-1500 range.

I'll run some experiments.

Reply:
Nope, I misunderstood, this is a tolerance window that makes the node more patient the further back it's starting. It's fine as-is.